design/one-pager-ignore-changes.md

```diff
@@ -0,0 +1,354 @@
+field is marked under `ignoreChanges`. Additionally, as Crossplane will soon
+support `status.atProvider`, users will be able to observe the actual values
+of the ignored fields.
+
+#### Ignoring required fields
+
+As there is no standard to distinguish which required fields are
+required for create and update and which just for creat, we can only
+advise to use caution when putting required fields into `ignoreChanges`.
+If such a need arises, it should preferably be handled as a special case
+on the provider side, and not through the `ignoreChanges` feature.
+
+#### Implementation
+
+The implementation would be done on the provider level, so we would need to
+update all the providers over time to use the new field.
+
+For providers using Upjet the solution will be to leverage Terraform's
+[ignore_changes] lifecycle field. So Upjet generation should be updated
+with a step that transfers the `spec.ignoreChanges` field into Terraform's
+`ignore_changes`. Something like this in the Upjet method that writes the
+main TF file:
+
+```go
+// WriteMainTF writes the content main configuration file that has the desired
+// state configuration for Terraform.
+func (fp *FileProducer) WriteMainTF() error {
+
+    fp.parameters["lifecycle"] = map[string]interface{}{
+        // If the resource is in a deletion process, we need to remove the
+        // deletion protection.
+        "prevent_destroy": !meta.WasDeleted(fp.Resource),
+        // Add fields which should be ignored on updates.
+        "ignore_changes": fp.Resource.GetIgnored(),
+    }
+...
+```
+
+For other providers we can implement a helper function in the
+[crossplane-runtime] repo that would use the existing code in
+[crossplane-runtime/fieldpath] to unset the fields of a managed resource
+that are under `ignoreChanges`. That way provider maintainers could use this
+function in the Update method without the need to invent their own solution.
+
+Helper function would look something this (WIP):
+```go
+func UnsetIgnored(managed resource.Managed) error {
+	p, err := PaveObject(managed)
+	if err != nil {
+		return err
+	}
+	
+	for _, s := range managed.GetIgnored() {
+		err = p.DeleteField(s)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+```
+
+#### Feature Gating
+
+Similar to all other new features being added to Crossplane, we will ship this
+new policy as an alpha feature that will be off by default and will be
+controlled by the `--enable-alpha-ignore-changes` flag in Providers.
+
+This will not prevent the field from appearing in the schema of the managed
+resources. However, we will ignore the `spec.ignoreChanges` when the feature
+is not enabled.
+
+#### Alternative feature gating approach
+
+To avoid changing the API, we could ship the `ignoreChanges` field as an
+optional annotation as a first solution. This would allow us to survey
+users and see how much this feature is useful before adding it to the API.
+
+### initProviders
```

Comment:
Should this still appear under Alternatives Considered? It appears to be part of the proposed approach.

Reply:
right, its not viable as a standalone solution for all the use cases so ill remove it.